### PR TITLE
AAU, on automl, when I launch prioritize.py, I want to output the list of the image embeddings

### PR DESCRIPTION
**Pull request by crsegerie**
Summary:
- Add the file prioritize.py
- Factorize the downloading function
- Memoize the downloaded images.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-*.pyc
\ No newline at end of file
+*.pyc
+.cachedir/**
```
Comment (PierreLeveau) [inline .gitignore:2]: no need if you use the automl's `.cache` dir (see below)

File: predict.py [hunk 2]
```diff
@@ -111,12 +127,18 @@ def main(
...
 
     kili = Kili(api_key=api_key)
     input_type, jobs = get_project(kili, project_id)
-    assets = get_assets(kili, project_id, label_types.split(","), max_assets = max_assets)
+    assets = get_assets(
+        kili,
+        project_id,
+        label_types.split(","),
+        max_assets=max_assets,
+        only_labeled=True,
```
Comment (PierreLeveau) [inline predict.py:140]: Good idea to let the possibility to select the unlabeled assets. However I think we should define how `label_types` and `only_labeled` are articulated. I propose the following args:
```
* get_labeled=True,
* get_unlabeled=True,
* label_types_in=["DEFAULT", "REVIEW"] (ignored if get_labeled = False)
```
if `get_unlabeled==False` and `get_labeled=False`, this should crash.
WDYT?

File: prioritize.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import os
+
+import click
```
Comment (PierreLeveau) [inline prioritize.py:3]: group the imports as follows:
1. python sys packges
2. python pip packages (including kili)
3. repo modules

File: prioritize.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    kili_print,
+    parse_label_types,
+)
+
+
+def embeddings_images(
+    images: list[PILImage],
+) -> np.ndarray:
+    """Get the embeddings of the images using a generic model trained on ImageNet."""
+    img2vec = Img2Vec(cuda=torch.cuda.is_available())
```
Comment (PierreLeveau) [inline prioritize.py:28]: what's the behavior if cuda is not available?

File: utils/helpers.py [hunk 5]
```diff
@@ -1,13 +1,23 @@
 import os
 from typing import List, Optional, Dict, Tuple
 from glob import glob
+from io import BytesIO
```
Comment (PierreLeveau) [inline utils/helpers.py:4]: same as above for grouping the imports

File: utils/helpers.py [hunk 6]
```diff
@@ -1,13 +1,23 @@
...
+
 from tqdm import tqdm
+from PIL import Image
+from PIL.Image import Image as PILImage
+import requests
+from tqdm.auto import tqdm
 
 from utils.constants import HOME
 
+memory = Memory(".cachedir")
```
Comment (PierreLeveau) [inline utils/helpers.py:19]: rather use the existing AutoML cache directory, for example `.cache/kili/automl/<project_id>/assets`

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
You can run a training procedure in this colab: https://colab.research.google.com/drive/1E-tyew18CESfV4w7P3ETbKo_WCxtJtlW?usp=sharing to test if there is no regression

File: utils/helpers.py [hunk 7]
```diff
@@ -118,3 +144,55 @@ def get_last_trained_model_path(job_name: str, project_id: str, model_path: str,
...
+        },
+    ).content
+
+    image = Image.open(BytesIO(img_data))
+    return image
+
+
+def download_project_images(
+    api_key, assets, inference_path: Optional[str] = None
+) -> list[DownloadedImages]:
```
Comment (PierreLeveau) [inline utils/helpers.py:172]: `list` -> `List`

File: prioritize.py (hunk 4 above)
Comment (crsegerie) [reply prioritize.py:28]: It works with the cpu. For example, it works on the mac

Comment (PierreLeveau) [reply prioritize.py:28]: ok, I misunderstood 😄

File: utils/helpers.py (hunk 7 above)
Comment (crsegerie) [reply utils/helpers.py:172]: Yes, colab uses python 3.7

File: predict.py (hunk 2 above)
Comment (crsegerie) [reply predict.py:140]: I agree, doing it

**Pull request closed without merging**